plugins/playstation/src/playstation-header.vala

```diff
@@ -0,0 +1,87 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStationHeader : Object {
+  private File cue_file;
+  private File bin_file;
+	private string parent_directory;
+
+	private const size_t[] OFFSETS = { 0xD368, 0xD3A8 };
+	private const string[] IDS = { "SLUS", "SCUS", "SLES", "SCES", "SLPS", "SLPM", "SCPS" };
+
+	private size_t? header_offset;
+
+  public PlayStationHeader (File cue_file) {
+    this.cue_file = cue_file;
+  }
+
+  public void check_validity () throws PlayStationError {
+    check_cue_info ();
+
+		foreach (var offset in OFFSETS) {
+			var data = get_string_at_offset (offset, 50);
```

Comment:
4 should be enough as it's the length of the magic value we want to check, actually it would be better to have "BOOT" as a private const stating it is our magic value, you could then just use `MAGIC_VALUE.length".